packages/api-graphql/src/internals/APIClient.ts

```diff
@@ -732,6 +732,15 @@ export function generateGraphQLDocument(
 			graphQLArguments ??
 				(graphQLArguments = {
 					filter: `Model${name}FilterInput`,
+					sortDirection: 'ModelSortDirection',
+					...[primaryKeyFieldName, ...sortKeyFieldNames].reduce(
+						(acc: Record<string, any>, fieldName) => {
+							acc[fieldName] = `${fields[fieldName].type}!`;
```

Comment:
The `!` after the type makes these required input fields for list operations. AppSync will reject this query if the PK/SK aren't present in variables. 

They need to be optional:
```suggestion
							acc[fieldName] = `${fields[fieldName].type}`;
```